### ISSUE TITLE
New navigation doesn't work with keyboard

### ISSUE THREAD

**Issue by alexgibson** [MozillaFoundation/foundation.mozilla.org#12292]
### Describe the bug

I just saw the new navigation on staging. It looks really nice! but I noticed it doesn't work with keyboard which is an a11y problem.

### To Reproduce
1. Go to https://foundation.mofostaging.net/
2. Press tab
3. Press tab again

### Expected behavior
1. On the second tab press when focus moves from the Mozilla word mark, the first navigation category link "Who we Are" should receive focus. 
2. On pressing the enter, the "Who we Are" menu should open and the first link within that menu should the receive focus. 
3. Pressing tab again should then move focus through the category links in the "Who we Are" menu.
4. When you tab through to the last link in the category, the "Who we Are" menu should close and move focus to "What We Do".

Instead, tab skips the navigation entirely and moves to the donate link.

### Desktop (please complete the following information):
 - OS: macOS
 - Browser Firefox Nightly
 - Version 127

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-609)

**Comment by TARishabh** [comment]
@alexgibson I guess we can't fix this issue, right? Because this repo is not linked to the website you mentioned

**Comment by jhonatan-lopes** [comment]
Hey @alexgibson, thanks for submitting this! I believe the team is aware of the issue, as it is covered by #12229.

**Comment by mmmavis** [comment]
Thanks all. Yes, fix is on the way!

**Comment by danielfmiranda** [comment]
Closed as completed via https://github.com/MozillaFoundation/foundation.mozilla.org/pull/12314!
